packages/truffle-contract/lib/contract.js

```diff
@@ -609,21 +723,25 @@ var contract = (function(module) {
       abi.forEach(function(item) {
         if (item.type != "event") return;
 
-        var signature = item.name + "(";
+        if (item.signature) {
```

Comment:
It seems the new web3 events have the signature already calculated, and for the tuple event it didn't match the one we calculated below. I didn't fully understand how this all worked, so I figured I'd best play it safe and branch on whether the `signature` is precalculated.